internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadWriteSchemaRepository.java

```diff
@@ -246,15 +246,14 @@ public SchemaEntry getSupersetOrLatestValueSchema(String storeName) {
   }
 
   @Override
-  public Optional<SchemaEntry> getSupersetSchema(String storeName) {
-    Optional<Integer> supersetSchemaID = getSupersetSchemaID(storeName);
-    return supersetSchemaID.map(schemaID -> accessor.getValueSchema(storeName, String.valueOf(schemaID)));
+  public SchemaEntry getSupersetSchema(String storeName) {
+    Integer supersetSchemaID = getSupersetSchemaID(storeName);
```

Comment:
Why don't we  have to check whether the `supersetSchemaID` is valid or not, just like some changes you made above?

Reply:
Actually I could short the lookup and return null if its invalid.